### PR TITLE
Add Manage Suppliers page

### PR DESCRIPTION
**Pull request by RahulJha-mastek**
Story AB#15247
Task AB#15460

### CODE REVIEW

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Views/Home/ManageSuppliers.cshtml [hunk 1]
```diff
@@ -0,0 +1,84 @@
+﻿@model IList<NHSD.GPIT.BuyingCatalogue.EntityFramework.Models.GPITBuyingCatalogue.Supplier>;
```
Comment (leerogers-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Views/Home/ManageSuppliers.cshtml:1]: This whole page needs the UI Components applying.
I can go through with you tomorrow if you want on how to apply these :)

**Review by leerogers-bjss — CHANGES_REQUESTED** [review]

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Views/Home/ManageSuppliers.cshtml (hunk 1 above)
Comment (RahulJha-mastek) [reply src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Views/Home/ManageSuppliers.cshtml:1]: Replaced HTML with UI components

**Review by leerogers-bjss — DISMISSED** [review]
hey rahul, as i mentioned on slack

all the <vc:nhs-action-link> url's need to be updated to use the @URL.Action() function for URL generation :)

**Comment by RahulJha-mastek** [comment]
> hey rahul, as i mentioned on slack
> 
> all the vc:nhs-action-link url's need to be updated to use the @URL.Action() function for URL generation :)


Updated vc:nhs-action-link url's with @URL.Action() in `Index.cshtml`. And in `ManageSuppliers.cshtml`, URL can be triggered with string format route URL.

File: tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Controllers/HomeControllerTests.cs [hunk 2]
```diff
@@ -94,6 +94,9 @@ public static async Task Get_BuyerOrganisations_MapsOrganisationsToModels()
             mockOrganisationService.Setup(o => o.GetAllOrganisations())
                 .ReturnsAsync(mockOrganisations);
             var mockMapper = new Mock<IMapper>();
+          
+            var mockSolutionService = new Mock<ISolutionsService>();
```
Comment (bjss-chris) [inline tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Controllers/HomeControllerTests.cs:98]: It looks like this variable isn't used.

File: tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Controllers/HomeControllerTests.cs [hunk 3]
```diff
@@ -115,7 +118,9 @@ public static async Task Get_BuyerOrganisations_ReturnsViewWithExpectedViewModel
             var mockOrganisationModels = new Mock<IList<OrganisationModel>>().Object;
             mockMapper.Setup(m => m.Map<IList<Organisation>, IList<OrganisationModel>>(mockOrganisations))
                 .Returns(mockOrganisationModels);
-            var controller = new HomeController(                
+            var mockSolutionService = new Mock<ISolutionsService>();
```
Comment (bjss-chris) [inline tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Controllers/HomeControllerTests.cs:121]: As above, it looks like this variable isn't used.

File: tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Controllers/HomeControllerTests.cs [hunk 4]
```diff
@@ -172,5 +177,51 @@ public static async Task Get_AddSolution_ReturnsViewWithExpectedViewModel()
...
+
+        [Fact]
+        public static void Get_ManageSuppliers_RouteAttribute_ExpectedTemplate()
+        {
+            typeof(HomeController)
+                .GetMethod(nameof(HomeController.ManageSuppliers))
+                .GetCustomAttribute<RouteAttribute>()
+                .Template.Should()
+                .Be("manage-suppliers");
+        }
```
Comment (bjss-chris) [inline tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Controllers/HomeControllerTests.cs:189]: I don't think this test adds any value (it's not testing behaviour).

**Review by bjss-chris — CHANGES_REQUESTED** [review]

File: tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Controllers/HomeControllerTests.cs (hunk 2 above)
Comment (RahulJha-mastek) [reply tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Controllers/HomeControllerTests.cs]: Removed unused variable.

File: tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Controllers/HomeControllerTests.cs (hunk 3 above)
Comment (RahulJha-mastek) [reply tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Controllers/HomeControllerTests.cs]: Removed unused variable.

File: tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Controllers/HomeControllerTests.cs (hunk 4 above)
Comment (RahulJha-mastek) [reply tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Controllers/HomeControllerTests.cs]: Removed test case.

**Review by bjss-chris — APPROVED** [review]

**Pull request merged**